akka-typed-tests/src/test/scala/akka/typed/cluster/ClusterApiSpec.scala

```diff
@@ -0,0 +1,74 @@
+/**
+ * Copyright (C) 2009-2017 Lightbend Inc. <http://www.lightbend.com>
+ */
+package akka.typed.cluster
+
+import akka.cluster.ClusterEvent._
+import akka.typed.TypedSpec
+import akka.typed.internal.adapter.ActorSystemAdapter
+import akka.typed.scaladsl.adapter._
+import akka.typed.testkit.TestKitSettings
+import akka.typed.testkit.scaladsl.TestProbe
+import com.typesafe.config.ConfigFactory
+import org.scalatest.concurrent.ScalaFutures
+
+import scala.concurrent.Await
+import scala.concurrent.duration._
+
+object ClusterApiSpec {
+  val config = ConfigFactory.parseString(
+    """
+      akka.actor.provider = cluster
+      akka.remote.artery.enabled = true
+      akka.remote.artery.canonical.port = 25552
+      akka.cluster.jmx.multi-mbeans-in-same-jvm = on
+      akka.actor {
+        serialize-messages = off
+        allow-java-serialization = off
+      }
+    """)
+}
+
+class ClusterApiSpec extends TypedSpec(ClusterApiSpec.config) with ScalaFutures {
+
+  val testSettings = TestKitSettings(adaptedSystem)
+  val clusterNode1 = Cluster(adaptedSystem)
+  val untypedSystem1 = ActorSystemAdapter.toUntyped(adaptedSystem)
+
+  val system2 = akka.actor.ActorSystem(
+    adaptedSystem.name,
+    ConfigFactory.parseString("akka.remote.artery.canonical.port = 25553").withFallback(ClusterApiSpec.config))
+  val adaptedSystem2 = system2.toTyped
+  val clusterNode2 = Cluster(adaptedSystem2)
+
+  object `A typed cluster` {
+
+    def `01 must join a cluster and observe events from both sides`() = {
+      val node1Probe = TestProbe[MemberEvent]()(adaptedSystem, testSettings)
+      clusterNode1.subscriptions ! Subscribe(node1Probe.ref)
```

Comment:
so it's picking up the class to subscribe to from the `ActorRef[MemberEvent]`, that's pretty cool